Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,179 @@ This document includes four main sections:
-
-   # Save (commit) the changes
    con.commit()
 
-   # We can also close the connection if we are done with it.
-   # Just be sure any changes have been committed or they will be lost.
-   con.close()
-
-The saved data is persistent: it can be reloaded in a subsequent session even
-after restarting the Python interpreter::
-
-   import sqlite3
-   con = sqlite3.connect('example.db')
-   cur = con.cursor()
+We can verify that the data was inserted correctly
+by executing a ``SELECT`` query.
+Use the now-familiar :meth:`cur.execute(...) <Cursor.execute>` to
+assign the result to ``res``,
+and call :meth:`res.fetchall() <Cursor.fetchall>` to return all resulting rows::
 
-At this point, our database only contains one row::
+   >>> res = cur.execute("SELECT score FROM movie")
+   >>> res.fetchall()
+   [(8.2,), (7.5,)]
 
-   >>> res = cur.execute('SELECT count(rowid) FROM stocks')
-   >>> print(res.fetchone())
-   (1,)
+The result is a :class:`list` of two :class:`!tuple`\s, one per row,
+each containing that row's ``score`` value.
 
-The result is a one-item :class:`tuple`:
-one row, with one column.
-Now, let us insert three more rows of data,
-using :meth:`~Cursor.executemany`::
+Now, insert three more rows by calling
+:meth:`cur.executemany(...) <Cursor.executemany>`::
 
-   >>> data = [
-   ...    ('2006-03-28', 'BUY', 'IBM', 1000, 45.0),
-   ...    ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0),
-   ...    ('2006-04-06', 'SELL', 'IBM', 500, 53.0),
-   ... ]
-   >>> cur.executemany('INSERT INTO stocks VALUES(?, ?, ?, ?, ?)', data)
+   data = [
+       ("Monty Python Live at the Hollywood Bowl", 1982, 7.9),
+       ("Monty Python's The Meaning of Life", 1983, 7.5),
+       ("Monty Python's Life of Brian", 1979, 8.0),
+   ]
+   cur.executemany("INSERT INTO movie VALUES(?, ?, ?)", data)
+   con.commit()  # Remember to commit the transaction after executing INSERT.
 
-Notice that we used ``?`` placeholders to bind *data* to the query.
+Notice that ``?`` placeholders are used to bind ``data`` to the query.
 Always use placeholders instead of :ref:`string formatting <tut-formatting>`
 to bind Python values to SQL statements,
-to avoid `SQL injection attacks`_.
-See the :ref:`placeholders how-to <sqlite3-placeholders>` for more details.
+to avoid `SQL injection attacks`_
+(see :ref:`sqlite3-placeholders` for more details).
 
-Then, retrieve the data by iterating over the result of a ``SELECT`` statement::
+We can verify that the new rows were inserted
+by executing a ``SELECT`` query,
+this time iterating over the results of the query::
 
-   >>> for row in cur.execute('SELECT * FROM stocks ORDER BY price'):
+   >>> for row in cur.execute("SELECT year, title FROM movie ORDER BY year"):
    ...     print(row)
+   (1971, "And Now for Something Completely Different")
+   (1975, "Monty Python and the Holy Grail")
+   (1979, "Monty Python's Life of Brian")
+   (1982, "Monty Python Live at the Hollywood Bowl")
+   (1983, "Monty Python's The Meaning of Life")
```

Comment:
Instead of repeating the same query below and duplicating the (longish) output, do you think it would be enough to do a `SELECT count` and show that there are now 5 movies?  This will make the tutorial shorter, introduce and teach a different query (the `count`), and in the final example you can show the full result.

Reply:
That is a good idea, but the point of this query is to introduce the cursor iterator. That would be moot if we change the query to return just one row. Perhaps we could alter the last example to use a more fancy output? Something like:

```python3
for row in cur.execute("SELECT title, year FROM movie ORDER BY year"):
    print(f'- "{title}" was released in {year}')
```

or:

```python3
releases = [r[0] for r in cur.execute("SELECT year FROM movie ORDER BY year")]
print(f"Monty Python released {len(releases)} movies between {releases[0]} and {releases[-1]}")
```

Reply:
I'm going to apply a variant of this:

```python3
res = cur.execute("SELECT title, year FROM movie ORDER BY score DESC")
title, year = res.fetchone()
print(f"The highest scoring Monty Python movie is {title}, released in {year}")
```

<details>
<summary>Non-scientific research: trying the tutorial out in real life</summary>

I tested this tutorial out on some folks I know; a scientist in hydrodynamics, and an electrical engineer. Both only know a bare minimum of Python (they know how to write short and simple scripts; most of their "programming" is modifying other people's scripts); they both have only a vague understanding of what a database is. Results:

- the scientist had no comments other than "this is a very good tutorial"
- the engineer commented that the `score` field was not used too much in the examples; other than that, it was remarked that this is a great improvement compared to the old tutorial

</details>

Reply:
A scientist, an engineer and a programmer walk into a bar...

> I'm going to apply a variant of this:

Only issue is this doesn't directly show the iterator behavior, no?

> I tested this tutorial out on some folks I know; a scientist in hydrodynamics, and an electrical engineer. 

Seems pretty "scientific" to me :laughing:

Reply:
> Only issue is this doesn't directly show the iterator behavior, no?

I'm talking about the very last example (verifying that the database was written to disk), not the iterator-example :)